### PR TITLE
Feature/abx 4286

### PR DESCRIPTION
**Pull request by Sam-Jeston**

### CODE REVIEW

File: src/components/LabelledField.tsx [hunk 1]
```diff
@@ -23,10 +24,11 @@ export const HorizontalLabelledField: React.SFC<Props> = (props) => (
       <label className='label'>{props.label}</label>
     </div>
     <div className='field-body'>
-      <div className='field'>
+      <div className={`field ${props.addon ? 'has-addons' : ''}`}>
```
Comment (Carl-Foster) [inline src/components/LabelledField.tsx]: Oooh no `&&`... controversial

File: src/components/PayeeList.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { Payee } from '@types'
+import * as React from 'react'
+
+type RemovePayee = (payeeName: string) => void
+export interface Props {
+  payees: Payee[]
+  removePayee: RemovePayee
+}
+
+export const deletePayeeLink: React.SFC<{name: string, removePayee: RemovePayee}> = ({name, removePayee}) => {
```
Comment (Carl-Foster) [inline src/components/PayeeList.tsx]: Since this is a component, it should be capitalised.

File: src/components/PayeeList.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+export const PayeeList: React.SFC<Props> = (props) => {
+  const payees = props.payees.map((payee) => {
+    return (
+      <HorizontalLabelledField
+        label={payee.name}
+        key={payee.name}
+        value={payee.publicKey}
+        addon={deletePayeeLink({name: payee.name, removePayee: props.removePayee})}
```
Comment (Carl-Foster) [inline src/components/PayeeList.tsx]: Once it is capitalised you would call it like `<DeletePayee removePayee={() => props.removePayee(payee.name)} />`

File: src/components/PayeeSet.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  validateProps = (): void | never => {
+    this.checkValidEntry('name')
+    this.checkValidEntry('publicKey')
+  }
+
+  checkValidEntry = (key: keyof Payee) => {
+    if (!this.props.payee[key]) {
+      throw new InputError(`${startCase(key)} is required`, `input-${kebabCase(key)}`)
```
Comment (Carl-Foster) [inline src/components/PayeeSet.tsx]: The second argument should be ``payee-``

File: src/store/reducers/payees.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    }
+  },
+})
+
+function handleChange(name: keyof Payee) {
+  return (state = '', action: RootAction) => {
+    switch (action.type) {
+      case getType(updatePayeeForm):
+        return action.payload.field === name ? action.payload.newValue : state
+      case getType(clearPayeeForm):
```
Comment (Carl-Foster) [inline src/store/reducers/payees.ts]: Any reason why you don't just listen for the `setPayee` action?

File: src/store/reducers/payees.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+export const payees = combineReducers<PayeeState, RootAction>({
+  form: combineReducers<Payee, RootAction>({
+    name: handleChange('name'),
+    publicKey: handleChange('publicKey'),
+  }),
+  payees: (state = [], action) => {
+    switch (action.type) {
+      case getType(setPayee):
+        return state.concat(action.payload)
```
Comment (Carl-Foster) [inline src/store/reducers/payees.ts]: This is a redux no-no. `{...state, {...action.payload}}`

Comment (Carl-Foster) [reply src/store/reducers/payees.ts]: or just `{...state, action.payload}`

Comment (Sam-Jeston) [reply src/store/reducers/payees.ts]: concat returns a new array, it doesn't mutate, so I disagree. But! updated

Comment (Carl-Foster) [reply src/store/reducers/payees.ts]: Yeah I wrote this and was like, wait I think this is okay.

**Pull request merged**